qds_sdk/cluster.py

```diff
@@ -216,6 +216,13 @@ def _parse_create_update(cls, args, action):
                              default=None,
                              help="Use hadoop1 instead of hadoop2. This is the default.")
 
+        spark = hadoop_group.add_mutually_exclusive_group()
+        spark.add_argument("--use-spark",
+                           dest="use_spark",
+                           action="store_true",
+                           default=None,
+                           help="Turn on spark for this cluster")
```

Comment:
I am going to start a discussion on crippling the UI too. That was written
before the current surge in types of clusters. We spoke about designing
something proper in a PR RG had started.

On Wed, Apr 29, 2015 at 1:05 AM, Rohit Agarwal <notifications@github.com>
wrote:

> In qds_sdk/cluster.py
> <https://github.com/qubole/qds-sdk-py/pull/103#discussion_r29279913>:
>
> > @@ -216,6 +216,13 @@ def _parse_create_update(cls, args, action):
> >                               default=None,
> >                               help="Use hadoop1 instead of hadoop2. This is the default.")
> >
> > +        spark = hadoop_group.add_mutually_exclusive_group()
> > +        spark.add_argument("--use-spark",
> > +                           dest="use_spark",
> > +                           action="store_true",
> > +                           default=None,
> > +                           help="Turn on spark for this cluster")
>
> Valid point.
>
> Though saying --dont-use-spark would show their intention pretty clearly.
> I mean we can do that from the UI by changing the cluster type - so
> wondering whether the SDK should be crippled in this regard.
>
> I don't have any strong opinions on this other than parity between UI and
> SDK. So, your decision.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qubole/qds-sdk-py/pull/103/files#r29279913>.
>



-- 
Rajat Venkatesh | Engg Lead
Qubole Inc | www.qubole.com